### ISSUE TITLE
"/anything" route only allows access by GET method

### ISSUE THREAD

**Issue by warren-bank** [kennethreitz/httpbin#361]
```bash
> curl -X POST -d 'a=1&b=2' https://httpbin.org/anything

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>405 Method Not Allowed</title>
<h1>Method Not Allowed</h1>
<p>The method is not allowed for the requested URL.</p>
```

```bash
> curl -X PUT -d 'a=1&b=2' https://httpbin.org/anything

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>405 Method Not Allowed</title>
<h1>Method Not Allowed</h1>
<p>The method is not allowed for the requested URL.</p>
```

notes:
* [code](https://github.com/kennethreitz/httpbin/blob/3463b5ec56795a32ac44bb3e6af8ff720198ce10/httpbin/core.py#L167)
* should also output: 'data'

**Comment by warren-bank** [comment]
maybe something like..

```python
@app.route('/anything', methods=['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'TRACE'])
@app.route('/anything/<path:anything>', methods=['GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'TRACE'])
def view_anything(anything=None):
    """Returns request data."""

    return jsonify(get_dict('url', 'args', 'headers', 'origin', 'method', 'form', 'data', 'files', 'json'))
```

**Comment by OJFord** [comment]
Because the `/:verb` routes don't allow path data, (e.g. `/get/123`) the combined effect with this issue is that there's no way (that I can see) to use httpbin to simulate a RESTy 'collection'.

If the `/anything` route allowed all methods, then one could `POST /anything`; `GET /anything/123`; DELETE `/anything/123`; etc. (Note I'm not suggesting that httpbin actually create resources, that's clearly out of its scope.)

**Comment by OJFord** [comment]
> Because the /:verb routes don't allow path data, (e.g. /get/123) the combined effect with this issue is that there's no way (that I can see) to use httpbin to simulate a RESTy 'collection'.

Wrong: `/status/:code` allows all methods, so valid HTTP status codes can be used as resource IDs.

**Comment by warren-bank** [comment]
Though it isn't my place to comment (for the author) on the purpose of this tool, it feels to me that it's entirely out of scope for the backend to allow the user to define their own custom RESTful routes and then parse data out of them.. to report that data in the response. None of which has anything at all to do with the issue at hand.

GET works fine (sans any special parsing logic):

```bash
curl -X GET https://httpbin.org/anything/foo/bar/baz
```

```json
{
  "args": {}, 
  "headers": {
    "Accept": "*/*", 
    "Connection": "close", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.47.0"
  }, 
  "method": "GET", 
  "origin": "...", 
  "url": "https://httpbin.org/anything/foo/bar/baz"
}
```

..still, my concern is being able to echo headers and data for all of the standard request methods. (PS: GET should also echo data.. there's nothing in the spec that says a server should ignore it)

..sorry for speaking out of turn; I just want this issue to stay on point.

**Comment by OJFord** [comment]
> I just want this issue to stay on point.

I'm sorry if you feel I was leading it astray: I only meant to offer an example use-case for why one might want to use the various methods on `/anything`.

> it feels to me that it's entirely out of scope for the backend to allow the user to define their own custom RESTful routes and then parse data out of them.. to report that data in the response.

I'm not sure I quite understand what you mean here. All I want to do is `POST /anything`; `{METHOD} /anything/:anything`. I explicitly said I was *not* suggesting any creation or tracking of the 'resources' created, and that that would be "clearly out of [httpbin]'s scope".

**Comment by warren-bank** [comment]
Apologies if I misunderstood your comments. Didn't mean to offend.

As for a concrete use-case. I found this tool the other day when I wanted to run a [few tests](https://github.com/warren-bank/node-request/blob/1fb396c7fb28e51afb1a178d7442c81ca1247b7e/tests/js/04.js#L58) on an [HTTP client library](https://github.com/warren-bank/node-request).. to make sure that it was sending the requests properly.

The default path was chosen to be "/anything". The issue described here is the reason that each test request is configured to override this value (with a specific ":verb"). It would've been preferable to not do so for 2 reasons:
* less configuration
* the response would repeat the requested method, which would be nice to include in the output log file. The workaround was to embed this information in the data stream, so it would appear in the log.
  * again: additional configuration
  * doesn't work for the `/get` route
    * workaround: embedded in querystring
    * again: additional configuration
